test/tools/scripts/extract-testcases.py

```diff
@@ -0,0 +1,227 @@
+#!/usr/bin/env python3
+
+import re
+import os
+
+
+class Test:
+    def __init__(self, name, content):
+        self.name = name
+        self.content = content
+        self.extractable = True
+        self.source = ""
+        self.comment = ""
+        self.tests = []
+        self.alsoViaYul = False
+
+    def analyse(self):
+        self.extractable = True
+        self.source = self.get_source()
+        self.content = self.content.replace(self.source, "", 1)
+
+        for not_allowed in ["m_contractAddress", "m_compiler", "m_evmHost",
+                            "m_revertStrings", "m_optimiserSettings", "gasLimit(", "gasPrice(", "blockNumber(",
+                            "blockTimestamp(",
+                            "numLogTopics(", "logTopic(", "numLogs(", "m_output", "m_sender", "sendMessage(",
+                            "m_transactionSuccessful", "BOOST_CHECK", "BOOST_REQUIRE",
+                            "solidity::test::CommonOptions::get()", 'bytes{',
+                            'bytes(', "testContractAgainstCppOnRange(", "testContractAgainstCpp(",
+                            "callContractFunctionWithValue(", ')YY\";']:
+            self.extractable &= (self.content.find(not_allowed) == -1)
+
+        self.extractable &= self.content.count("compileAndRun") == 1
+        self.extractable &= len(self.source) > 0
+        self.alsoViaYul = self.content.find("ALSO_VIA_YUL") != -1
+        for line in self.content.split("\n"):
+            line = line.strip()
+            if line.startswith("ABI_CHECK(") & (not line.startswith("ABI_CHECK(callContractFunction")):
```

Comment:
This means you ignore lines that do not start with `ABI_CHECK(`? Can you print what these lines are?

Reply:
Good idea

Reply:
Output can be found here https://gist.github.com/aarlt/317fa5c877a702fddd348a80f95b7e14#file-extractor_not_matching_lines

Reply:
Not matching regexp can be found here https://gist.github.com/aarlt/317fa5c877a702fddd348a80f95b7e14#file-extractor_not_matching_regexp

I ran the extractor ton the original `SolidityEndToEndTest.cpp`